esp-hal/src/dma/mod.rs

```diff
@@ -573,6 +588,32 @@ macro_rules! dma_circular_descriptors_chunk_size {
     };
 }
 
+/// Convenience macro to create a DmaTxBuf from buffer size. The buffer and
+/// descriptors are statically allocated and used to create the `DmaTxBuf`.
+///
+/// ## Usage
+/// ```rust,no_run
+#[doc = crate::before_snippet!()]
+/// use esp_hal::dma_tx_buffer;
+/// use esp_hal::dma::DmaBufBlkSize;
+///
+/// let tx_buf =
+///     dma_tx_buffer!(32000);
+/// # }
+/// ```
+#[macro_export]
+macro_rules! dma_tx_buffer {
+    ($tx_size:expr) => {{
+        const TX_DESCRIPTOR_LEN: usize =
+            $crate::dma::DmaTxBuf::compute_descriptor_count($tx_size, None);
+        static mut TX_BUFFER: [u8; $tx_size] = [0u8; $tx_size];
```

Comment:
This requires the round-up-and-align trick.

Reply:
Not on the TX side, strictly speaking

Reply:
It does I think for the esp32

Reply:
Ahh yes, the esp32 wants word aligned sizes, but the round-up-and-align trick doesn't fix that. You'd just have to add an assert that tx_size is a multiple of 4 on the esp32.